src/traces/scatter/calc.js

```diff
@@ -115,6 +115,10 @@ module.exports = function calc(gd, trace) {
     for(i = 0; i < serieslen; i++) {
         cd[i] = (isNumeric(x[i]) && isNumeric(y[i])) ?
             {x: x[i], y: y[i]} : {x: false, y: false};
+
+        if(trace.identifier && trace.identifier[i] !== undefined) {
```

Comment:
Should we sanity-check more than just `undefined`? Maybe we should allow only strings and number? So maybe, casting every item to a string (i.e. `String(trace.indentifiers[i])`) is the way to go here.